src/Model/ProfileField.php

```diff
@@ -62,9 +62,7 @@ public function __get($name)
 
 		switch ($name) {
 			case 'permissionset':
-				$this->permissionset =
-					$this->permissionset ??
-						$this->permissionSetRepository->selectFirst(['id' => $this->psid, 'uid' => $this->uid]);
+				$this->permissionset = $this->permissionset ?? $this->permissionSetDepository->selectOneById($this->psid);
```

Comment:
The previous code was checking for ownership, you need to maintain this feature parity with an additional check on the entity field value or in a separate Depository method.